site/content/en/latest/contributions/design/backend.md

```diff
@@ -0,0 +1,151 @@
+kind: HTTPRoute
+metadata:
+  name: backend
+spec:
+  parentRefs:
+    - name: eg
+  hostnames:
+    - "www.example.com"
+  rules:
+    - backendRefs:
+        - group: gateway.envoyproxy.io
+          kind: Backend
+          name: backend-mixed-ip-uds
+          weight: 1
+        - group: ""
+          kind: Service
+          name: backend
+          port: 3000
+          weight: 1          
+      matches:
+        - path:
+            type: PathPrefix
+            value: /
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: BackendTLSPolicy
+metadata:
+  name: policy-btls
+spec:
+  targetRef:
+    group: gateway.envoyproxy.io
+    kind: Backend
+    name: backend-mixed-ip-uds
+    sectionName: uds-be
+  tls:
+    caCertRefs:
+      - name: backend-tls-checks-certificate
+        group: ''
+        kind: ConfigMap
+    hostname: example.com
+```
+
+## Design Decisions
+* All existing and future `BackendObjectReference` in Envoy Gateway MUST support the `Backend` kind. 
+* Gateway-API and Envoy Gateway policies that attach to Services ([BackendTLSPolicy][], [BackendLBPolicy][]) 
+  MUST support attachment to the `Backend` resource in Envoy Gateway. Policies may attach to a named 
+  section of the `Backend` resource (the `backendAddress.name` field). 
+* The `Backend` API SHOULD support other Gateway-API backend features, such as [Backend Protocol Selection][]. 
+  Translation of explicit upstream application protocol setting MUST be consistent with the existing implementation for
+  `Service` resources. 
+* This API resource MUST be part of same namespace as the targetRef resource. The `Backend` API MUST be subject to 
+  the same cross-namespace reference restriction as referenced `Service` resources.    
+* The `Backend` resource translation MUST NOT modify Infrastructure. Any change to infrastructure that is required to 
+  achieve connectivity to a backend (mounting a socket, adding a sidecar container, modifying a network policy, ...) 
+  MUST be implemented with an appropriate infrastructure patch in the [EnvoyProxy][] API. 
+* To limit the overall maintenance effort related to supporting of non-Service backends, the `Backend` API SHOULD 
+  support multiple generic address types (UDS, FQDN, IPv4, IPv6), and MUST NOT support vendor-specific backend types.
+* Both `Backend` and `Service` resources may appear in the same `BackendRefs` list.
+* The Optional `Port` field SHOULD NOT be evaluated when referencing a `Backend`.  
+* Referenced `Backend` resources MUST be translated to envoy endpoints, similar to the current `Service` translation.
+* Certain combinations of `Backend` and `Service` are incompatible. For example, a Unix Domain Socket and a FQDN service
+  require different cluster service discovery types (Static/EDS and Strict-DNS accordingly).
+* If a Backend that is referenced by a route cannot be translated, the `Route` resource will have an `Accepted=False` 
+  condition with a `UnsupportedValue` reason. 
+  
```

Comment:
is there a valid use to supports UDS addresses in HTTPRoute ? I can't think of any

Reply:
If the gateway is deployed in a very coupled manner with the backend, it could make sense. For example, if both the backend(s) and gateway are daemonsets, they may communicate over a host-level UDS. 

Proxies do support this in general: [nginx](https://nginx.org/en/docs/http/ngx_http_upstream_module.html#upstream), [haproxy](https://www.haproxy.com/documentation/haproxy-configuration-manual/latest/#4.2-server)

Reply:
As discussed in the community meeting, we can choose to reject UDS for xRoute BackendRefs. The External Service use case is still valuable here.